kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/years/months/YearsMonthsValueConverterTest.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    @Test
+    public void testFromDMNStringYearsMonthsMultipleDigits() {
+
+        final String input = "duration(\"P12Y11M\")";
+        testFromDMNString(input, "12", "11");
+    }
+
+    @Test
+    public void testFromDMNStringYearsMonthsOnlyYearSingleDigit() {
+
+        final String input = "duration(\"P1Y\")";
+        testFromDMNString(input, "1", "");
+    }
+
+    @Test
+    public void testFromDMNStringYearsMonthsOnlyYear() {
+
+        final String input = "duration(\"P12Y\")";
+        testFromDMNString(input, "12", "");
+    }
+
+    @Test
+    public void testFromDMNStringYearsMonthsOnlyMonthSingleDigit() {
+
+        final String input = "duration(\"P1M\")";
+        testFromDMNString(input, "", "1");
+    }
+
+    @Test
+    public void testFromDMNStringYearsMonthsOnlyMonth() {
+
+        final String input = "duration(\"P12M\")";
+        testFromDMNString(input, "", "12");
+    }
+
+    @Test
+    public void testToDMNStringEmpty() {
+
+        testToDMNString("", "", "");
+    }
+
+    @Test
+    public void testToDMNStringOnlyMonth() {
+
+        testToDMNString("", "1", "duration(\"P1M\")");
+    }
+
+    @Test
+    public void testToDMNStringOnlyYear() {
+
+        testToDMNString("1", "", "duration(\"P1Y\")");
+    }
+
+    @Test
+    public void testToDMNStringYearAndMonth() {
+
+        testToDMNString("2", "1", "duration(\"P2Y1M\")");
+    }
+
+    private void testToDMNString(final String years,
+                                 final String months,
+                                 final String expected) {
+
+        final String actual = converter.toDMNString(years, months);
+
+        assertEquals(expected, actual);
+        verify(converter).matchSigns(any());
+    }
+
+    private void testFromDMNString(final String input,
+                                   final String expectedYears,
+                                   final String expectedMonths) {
+
+        final YearsMonthsValue actual = converter.fromDMNString(input);
+
+        assertEquals(expectedYears, actual.getYears());
+        assertEquals(expectedMonths, actual.getMonths());
+    }
+
+    @Test
+    public void testToDisplayOnlyMonths() {
+
+        final String expected = "1 mos";
+        final YearsMonthsValue value = new YearsMonthsValue();
+        value.setMonths("1");
+
+        testToDisplayValue(value, expected);
+    }
+
+    @Test
+    public void testToDisplayOnlyYears() {
+
+        final String expected = "1 yrs";
+        final YearsMonthsValue value = new YearsMonthsValue();
+        value.setYears("1");
+
+        testToDisplayValue(value, expected);
+    }
+
+    @Test
+    public void testToDisplayYearsAndMonths() {
+
+        final String expected = "1 years, 2 months";
+        final YearsMonthsValue value = new YearsMonthsValue();
+        value.setYears("1");
+        value.setMonths("2");
+
+        testToDisplayValue(value, expected);
+    }
+
+    @Test
+    public void testToDisplayYearsAndMonthsEmtpy() {
```

Comment:
I think there's a typo here ^
```suggestion
    public void testToDisplayYearsAndMonthsEmpty() {
```